### PR TITLE
✨ add accounts to the aws organization resource, --organization discovery

### PR DESCRIPTION
**Pull request by vjeffrey**
```
go run apps/cnquery/cnquery.go shell aws --discover organization --role arn:aws:iam::ACCOUNTID:role/test-org-list-accounts-role
```

```
cnquery> aws.account {  organization {accounts }}
aws.account: {
  organization: {
    accounts: [
      0: aws.account id="8913...."
      1: aws.account id="0724...."
```
Screenshot 2024-05-09 at 21 03 29

```
→ connected to AWS Account
  ___ _ __   __ _ _   _  ___ _ __ _   _
 / __| '_ \ / _` | | | |/ _ \ '__| | | |
| (__| | | | (_| | |_| |  __/ |  | |_| |
 \___|_| |_|\__, |\__,_|\___|_|   \__, |
  mondoo™      |_|                |___/  interactive shell

cnquery> aws.account.id
aws.account.id: "8913..."
```

### CODE REVIEW

File: providers/aws/resources/aws.lr.manifest.yaml [hunk 1]
```diff
@@ -2067,6 +2067,8 @@ resources:
       - aws
   aws.organization:
     fields:
+      accounts:
+        min_mondoo_version: latest
```
Comment (tas50) [inline providers/aws/resources/aws.lr.manifest.yaml]: 9.0

File: providers/aws/resources/aws_account.go [hunk 2]
```diff
@@ -52,3 +51,50 @@ func (a *mqlAwsAccount) organization() (*mqlAwsOrganization, error) {
...
+		return args, nil, nil
+	}
+	if len(args) == 0 {
+		if ids := getAssetIdentifier(runtime); ids != nil {
+			id := strings.TrimPrefix(ids.arn, "arn:aws:sts::")
+			args["id"] = llx.StringData(id)
+		}
+	}
+	if args["id"] == nil {
+		return args, nil, errors.New("nil account id")
```
Comment (czunker) [inline providers/aws/resources/aws_account.go]: nit: I'm not sure users understand "nil", when they see this error. Just use "no" instead, or "no ... specified"? Or ...

**Review by czunker — APPROVED** [review]
Thanks @vjeffrey

File: providers/aws/resources/aws_account.go (hunk 2 above)
Comment (vjeffrey) [reply providers/aws/resources/aws_account.go]: true true, ill change that

**Pull request merged**